axelar-chains-config/src/utils/sourcify.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+const axios = require('axios');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+async function findValue(obj, contractName, result = []) {
+    if (obj && typeof obj === 'object') {
+        for (const key in obj) {
+            if (key.includes(`/${contractName}.sol`) || key === contractName) {
+                result.push(obj[key].metadata ? obj[key].metadata : obj[key].content);
+            }
+
+            await findValue(obj[key], contractName, result);
+        }
+    }
+}
+
+async function readAndParseFile(filePath, targetKey) {
+    try {
+        const jsonData = await fs.promises.readFile(filePath, { encoding: 'utf8' });
+        const parsedData = JSON.parse(jsonData);
+        const result = [];
+        await findValue(parsedData, targetKey, result);
+        return result;
+    } catch (error) {
+        console.error(`Error reading or parsing file ${filePath}: ${error.message}`);
+        return [];
+    }
+}
+
+async function processDirectory(directoryPath, targetKey) {
+    try {
+        const fileNames = await fs.promises.readdir(directoryPath);
+        const filePaths = fileNames.map((fileName) => path.join(directoryPath, fileName));
+
+        const results = await Promise.all(filePaths.map((filePath) => readAndParseFile(filePath, targetKey)));
+
+        const uniqueResults = [...new Set(results.flat().filter(Boolean))];
+        return uniqueResults;
+    } catch (error) {
+        console.error(`Error reading directory ${directoryPath}: ${error.message}`);
+        return [];
+    }
+}
+
+function searchContractName(rootFolder, targetValue) {
+    const files = fs.readdirSync(rootFolder);
+
+    for (const fileOrFolder of files) {
+        const fullPath = path.join(rootFolder, fileOrFolder);
+
+        if (fs.statSync(fullPath).isDirectory()) {
+            const contractName = searchContractName(fullPath, targetValue);
+
+            if (contractName) {
+                return contractName;
+            }
+        } else if (fileOrFolder.endsWith('.json')) {
+            const jsonData = JSON.parse(fs.readFileSync(fullPath, 'utf-8'));
+
+            if (jsonData.deployedBytecode === targetValue) {
+                return jsonData.contractName;
+            }
+        }
+    }
+}
+
+async function verify(dir, provider, address, chainId) {
```

Comment:
```suggestion
async function verifySourcify(dir, provider, address, chainId) {
```
or something similar to distinguish it from other verify methods